src/utils/mpc-service.ts

```diff
@@ -0,0 +1,59 @@
+import { Signature } from '@near-js/crypto';
+import { serialize } from 'borsh';
+import { sha256 } from 'js-sha256';
+
+import { ConstructSignature } from './types';
+import { SCHEMA } from '@near-js/transactions';
+
+const hashToken = (oidcToken: string): number[] => {
+  const tokenHash = sha256.create();
+  tokenHash.update(oidcToken);
+  return tokenHash.array();
+};
+
+export const getUserCredentialsFrpSignature = ({
```

Comment:
This function should not have `shouldHashToken` parameter. You never need to hash OIDC Token in `user_credentials` request.

For some reason you are using this function in claim OIDC flow, where we need to hash it, because OIDC Token should not be exposed to the server at that step.
You should create a separate function with hashing for claiming flow.